__init__.py

```diff
@@ -0,0 +1,326 @@
+        """
+        self._handle_future_time(request)
+
+    @intent_handler("what-time-will-it-be.intent")
+    def handle_future_time_padatious(self, request: Message):
+        """Respond to a less common request for the future time.
+
+        Example: "when is it 8 hours from now?"
+
+        Args:
+            request: Data from the intent parser regarding the user's voice request.
+        """
+        self._handle_future_time(request)
+
+    @intent_handler(AdaptIntent("mark-one-idle").require("display").require("time"))
+    def handle_show_time(self, _):
+        """Respond to a request top show the time on a Mark I when idle.
+
+        Example: "What time will it be in 8 hours?"
+        """
+        self.settings["show_time"] = True
+        self.schedule_repeating_event(self.display_idle, None, 10, name="IdleTime")
+
+    def _handle_future_time(self, request: Message):
+        """Respond to a request for the future time.
+
+        Determine the future time, then speak the result to the user and display
+        it (if applicable).
+
+        Args:
+            request: Data from the intent parser regarding the user's voice request.
+        """
+        response = Response(self.config_core, self.location_regex_path)
+        try:
+            response.build_future_time_response(request.data["utterance"])
+        except LocationNotFoundError:
+            self._handle_location_not_found(response)
+        else:
+            if response.date_time is None:
+                self._handle_current_time(request)
+            else:
+                self._respond(response)
+
+    def _handle_current_time(self, request: Message):
+        """Respond to a request for the current time.
+
+        Determine the current time, then speak the result to the user and display
+        it (if applicable).
+
+        Args:
+            request: Data from the intent parser regarding the user's voice request.
+        """
+        self.log.info("request data: " + str(request.data.keys()))
+        response = Response(self.config_core, self.location_regex_path)
+        try:
+            response.build_current_time_response(request.data["utterance"])
+        except LocationNotFoundError:
+            self._handle_location_not_found(response)
+        else:
+            self._respond(response)
+
+    def _handle_location_not_found(self, response):
+        """User requested time in a city not recognized by a Geolocation API call."""
+        dialog_data = dict(location=response.requested_location)
+        self.speak_dialog("location-not-found", dialog_data)
+
+    def _respond(self, response: Response):
+        """Speak and display the response to the user's request.
+
+        Args
+            response: object used to formulate the response
+        """
+        self._display_time(response)
+        self.speak_dialog(response.dialog_name, response.dialog_data, wait=True)
+        time.sleep(10)
```

Comment:
if this is just to avoid clearing the screen use the override_idle parameter from the SkillGUI class, you can specify a number of seconds in there and let the rest of the stack handle things like it should. I agree with @ken-mycroft this is an ugly pattern. 

Since this is a full rewrite with no functional changes, let's aim to not have any code smells. 

If for some reason override_idle doesnt work at least leave a # TODO comment before the sleep call, most IDEs will flag those and hopefully someone will fix it

Reply:
oh, if this is because of the mark1 then you can at least move it to the proper place, or set a Timer thread.  

the skill handler completed event is only sent after the intent returns, so anything waiting for that event (to know that interaction ended) will suffer from that hardcoded sleep and potentially timeout

Reply:
Made some changes that I hope address your concerns.  Let me know if this satisfies your suggestion.

Reply:
I haven't heard back and Ken approved.  Merging now

Reply:
only had to time to check it out now, but yes, that was exactly what i meant :+1: